### PR TITLE
add japanese user login and signup to mercury tests

### PR DESCRIPTION
**Pull request by bekcunning**
https://wikia-inc.atlassian.net/browse/SOC-1163

add japanese user login and signup tests to mercury login/reg test cases.

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Define a constant instead of duplicating this literal "@gmail.com" 5 times.](https://github.com/Wikia/selenium-tests/blob/245449c3d72810d7f94a00d1ce4242c01f04b97b/src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java#L21) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "qaart001+" 5 times.](https://github.com/Wikia/selenium-tests/blob/245449c3d72810d7f94a00d1ce4242c01f04b97b/src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java#L21) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java [hunk 1]
```diff
@@ -75,4 +75,17 @@ public void MercurySignupTest_005_signupErrorTooYoungUser() {
           .typeBirthdate("12", "12", "2009").register();
     mobileSignup.verifyBirthdateError();
   }
+
+  @Test(groups = {"MercurySignupTest_006"})
+  @Execute(onWikia = "ja.ja-test")
+  public void MercurySignupTest_006_japaneseUserSignup() {
```
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java]: All tests in `SignupTests.java` have the same steps except last verification step.
Can we move those steps to separate method and call it in each test?

One more thing: `MercurySignupTest_006_japaneseUserSignup` and `MercurySignupTest_001_successfulSignup` tests have the same `verifyAvatarAfterSignup` method. Can we combine these tests into one and pass different data from TestNG DataProvider?

Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java]: There are some small differences, like different user, date etc. IMO tests looks clearer in that way

**Comment by tomasznapieralski** [comment]
All 6 tests are passing, LGTM

File: src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java [hunk 2]
```diff
@@ -195,8 +195,8 @@ public void SignUp_007_signUpWithFacebook() {
   public void SignUp_008_signupJapaneseUser() {
     SignUpPageObject signUp = new WikiBasePageObject(driver).navigateToSpecialSignUpPage(wikiURL);
     signUp.disableCaptcha();
-    String userName = "品質管理" + signUp.getTimeStamp();
-    String password = "品質管理管理" + signUp.getTimeStamp();
+    String userName = "ユーザー" + signUp.getTimeStamp();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java]: MAJOR Rename "userName" which hides the field declared at line 36. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java [hunk 3]
```diff
@@ -195,8 +195,8 @@ public void SignUp_007_signUpWithFacebook() {
...
+    String password = "パス" + signUp.getTimeStamp();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java]: MAJOR Rename "password" which hides the field declared at line 37. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues:
* MAJOR 10 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Define a constant instead of duplicating this literal "@gmail.com" 5 times.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java#L21) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "qaart001+" 5 times.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java#L21) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Rename "password" which hides the field declared at line 37.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L121) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Rename "password" which hides the field declared at line 37.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Remove this unused "jaTestWiki" private field.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L38) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Rename "userName" which hides the field declared at line 36.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L120) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Rename "password" which hides the field declared at line 37.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L145) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Rename "userName" which hides the field declared at line 36.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Rename "userName" which hides the field declared at line 36.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java#L144) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [8 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/bdbab8e16d0504af927abca48e8ed8d28022d190/src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java (hunk 1 above)
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SignupTests.java]: Yep, the small differences would come as method arguments, however don't agree about clearness.

**Comment by drets** [comment]
It's sad that people don't try to fix Sonar issues :disappointed:

**Pull request merged**